operator/v1alpha1/types_csi.go

```diff
@@ -0,0 +1,64 @@
+package v1alpha1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCSIDriver object allows management and configuration of a CSI driver operator
+// installed by default in OpenShift.
+type ClusterCSIDriver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterCSIDriverSpec `json:"spec"`
+
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterCSIDriverStatus `json:"status"`
+}
+
+// CSIDriverName is the name of the CSI driver
+type CSIDriverName string
+
+const (
+	AWSEBSCSIDriver CSIDriverName = "ebs.csi.aws.com-ebs"
+	ManilaCSIDriver CSIDriverName = "manila.csi.openstack.org"
+	OvirtCSIDriver  CSIDriverName = "csi.ovirt.org"
+)
+
+// ClusterCSIDriverSpec is the desired behavior of CSI driver operator
+type ClusterCSIDriverSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	DriverConfig            CSIDriverConfig `json:"driverConfig"`
```

Comment:
optional or required?

Reply:
fixed. amazingly though kubebuilder is smart enough to recognize that, a child field of this object is required and hence parent MUST be required too and marks `DriverConfig` field in generated CRD as required.